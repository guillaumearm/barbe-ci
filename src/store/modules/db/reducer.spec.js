const reducer = require('./reducer');

describe('store:db/reducer', () => {
  const unknownAction = { type: 'UNKNOWN' };
  const initialState = reducer(undefined, unknownAction);
  it('returns initial state', () => {
    expect(reducer(undefined, unknownAction)).toMatchSnapshot();
  });
  it('does not update state on unknown action', () => {
    expect(reducer(initialState, unknownAction)).toBe(initialState);
  });
  it('does not update state when [LOAD_DB] action payload is empty', () => {
    expect(reducer(initialState, { type: 'LOAD_DB' })).toBe(initialState);
    expect(reducer(initialState, { type: 'LOAD_DB', payload: {} })).toBe(initialState);
  });
  test('load db when receive a valid [LOAD_DB] action', () => {
    const newDb = { ...initialState, users: { user1: {} } }
    const loadDb = (db) => ({ type: 'LOAD_DB', payload: { db } });
    expect(reducer(initialState, loadDb(newDb))).toEqual(newDb);
  });
});
