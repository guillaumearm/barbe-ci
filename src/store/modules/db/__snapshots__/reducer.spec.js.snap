// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store:db/reducer returns initial state 1`] = `
Object {
  "commits": Object {},
  "repositories": Object {},
  "tokens": Object {},
  "users": Object {},
}
`;
